### PR TITLE
Fix index-state syntax

### PR DESCRIPTION
**Pull request by andreabedini**
# Description

The second index-state stanza completely ovverides the first, resetting hackage index state to HEAD. See haskell/cabal#8568

# Checklist

- Branch
    - [ ] Commit sequence broadly makes sense
    - [ ] Commits have useful messages
    - [ ] The documentation has been properly updated
    - [ ] New tests are added if needed and existing tests are updated
    - [ ] If this branch changes Consensus and has any consequences for downstream repositories or end users, said changes must be documented in [`interface-CHANGELOG.md`](../ouroboros-consensus/docs/interface-CHANGELOG.md)
    - [ ] If this branch changes Network and has any consequences for downstream repositories or end users, said changes must be documented in [`interface-CHANGELOG.md`](../docs/interface-CHANGELOG.md)
    - [ ] If serialization changes, user-facing consequences (e.g. replay from genesis) are confirmed to be intentional.
- Pull Request
    - [ ] Self-reviewed the diff
    - [ ] Useful pull request description at least containing the following information:
      - What does this PR change?
      - Why these changes were needed?
      - How does this affect downstream repositories and/or end-users?
      - Which ticket does this PR close (if any)? If it does, is it [linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)?
    - [ ] Reviewer requested

### CODE REVIEW

File: cabal.project [hunk 1]
```diff
@@ -12,10 +12,13 @@ repository cardano-haskell-packages
 
 -- See CONTRIBUTING for some Nix commands you will need to run if you
 -- update either of these.
--- Bump this if you need newer packages from Hackage
-index-state: 2022-12-01T00:00:00Z
--- Bump this if you need newer packages from CHaP
-index-state: cardano-haskell-packages 2022-11-15T00:00:00Z
+-- repeat the index-state for hackage to work around haskell.nix parsing limitation
+index-state: hackage.haskell.org 2022-12-01T00:00:00Z
```
Comment (amesgen) [inline cabal.project:16]: From https://github.com/haskell/cabal/issues/8568, it seems to me that we could delete this given the immediately succeeding multi-line `index-state` declaration, right?
```suggestion
```

File: cabal.project [hunk 2]
```diff
@@ -12,10 +12,13 @@ repository cardano-haskell-packages
...
+index-state: 
+  -- Bump this if you need newer packages from Hackage
+  , hackage.haskell.org 2022-12-01T00:00:00Z
```
Comment (amesgen) [inline cabal.project:19]: Ah, Nix CI is failing due to our hackage.nix pin being too old, which was only possible to pass CI due to the bug you are fixing here. So bumping hackage.nix should get CI on green :+1:

File: cabal.project (hunk 1 above)
Comment (andreabedini) [reply cabal.project:16]: ah damn it, that's supposed to be
```
index-state: 2022-12-01T00:00:00Z
```
too much copy and paste :disappointed:

That line is only useful for haskell.nix, cabal will override it with the following one.

**Review by coot — APPROVED** [review]
LGTM

File: cabal.project (hunk 2 above)
Comment (amesgen) [reply cabal.project]: There are now CI failures due to the new HLS version, maybe revert hackage.nix index state back to the first of December to unblock this PR?

**Comment by andreabedini** [comment]
bors try

**Comment by andreabedini** [comment]
bors try

**Comment by andreabedini** [comment]
bors try

**Comment by amesgen** [comment]
bors merge

**Comment by amesgen** [comment]
bors cancel

**Comment by amesgen** [comment]
bors merge

**Comment by amesgen** [comment]
bors merge

**Pull request merged**